File: evals/behavioral/tasks/typescript/fixtures/total-safe-div.bad.ts
// Not total: returns a bare number. In JS, 1 / 0 is Infinity and 0 / 0 is NaN —
// no error, no signal — so the caller gets a "usable" number and is never
// forced to consider the missing answer; `safeDiv(1, 0) + 1` type-checks.
export function safeDiv(a: number, b: number): number {
  return a / b;
}

File: evals/behavioral/tasks/python/fixtures/total-safe-div.bad.py
# Not total: returns a bare int and raises ZeroDivisionError at runtime for
# b == 0. The caller gets a "usable" number and is never forced to consider the
# missing answer, so `safe_div(1, 0) + 1` type-checks.
def safe_div(a: int, b: int) -> int:
    return a // b
